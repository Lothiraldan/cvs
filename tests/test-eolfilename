#!/bin/sh
# http://mercurial.selenic.com/bts/issue352

"$TESTDIR/hghave" eol-in-paths || exit 80

echo % test issue352
hg init foo
cd foo

A=`printf 'he\rllo'`

echo foo > "$A"
hg add
hg ci -A -m m
rm "$A"

echo foo > "hell
o"
hg add
hg ci -A -m m

echo foo > "$A"
hg debugwalk

# http://mercurial.selenic.com/bts/issue2036
cd ..
echo % test issue2039

hg init bar
cd bar

echo "[extensions]" >> $HGRCPATH
echo "color=" >> $HGRCPATH

A=`printf 'foo\nbar'`
B=`printf 'foo\nbar.baz'`

touch "$A"
touch "$B"

hg status --color=always

exit 0
