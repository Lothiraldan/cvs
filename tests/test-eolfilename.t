http://mercurial.selenic.com/bts/issue352

  $ "$TESTDIR/hghave" eol-in-paths || exit 80

test issue352

  $ hg init foo
  $ cd foo
  $ A=`printf 'he\rllo'`
  $ echo foo > "$A"
  $ hg add
  adding hello
  abort: '\n' and '\r' disallowed in filenames: 'he\rllo'
  [255]
  $ hg ci -A -m m
  adding hello
  abort: '\n' and '\r' disallowed in filenames: 'he\rllo'
  [255]
  $ rm "$A"
  $ echo foo > "hell
  > o"
  $ hg add
  adding hell
  o
  abort: '\n' and '\r' disallowed in filenames: 'hell\no'
  [255]
  $ hg ci -A -m m
  adding hell
  o
  abort: '\n' and '\r' disallowed in filenames: 'hell\no'
  [255]
  $ echo foo > "$A"
  $ hg debugwalk
  f  hello  hello
  f  hell
  o  hell
  o

http://mercurial.selenic.com/bts/issue2036

  $ cd ..

test issue2039

  $ hg init bar
  $ cd bar
  $ echo "[extensions]" >> $HGRCPATH
  $ echo "color=" >> $HGRCPATH
  $ A=`printf 'foo\nbar'`
  $ B=`printf 'foo\nbar.baz'`
  $ touch "$A"
  $ touch "$B"
  $ hg status --color=always
  [0;35;1;4m? foo[0m
  [0;35;1;4mbar[0m
  [0;35;1;4m? foo[0m
  [0;35;1;4mbar.baz[0m
