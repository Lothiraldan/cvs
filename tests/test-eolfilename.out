% test issue352
adding hello
abort: '\n' and '\r' disallowed in filenames: 'he\rllo'
adding hello
abort: '\n' and '\r' disallowed in filenames: 'he\rllo'
adding hell
o
abort: '\n' and '\r' disallowed in filenames: 'hell\no'
adding hell
o
abort: '\n' and '\r' disallowed in filenames: 'hell\no'
f  hello  hello
f  hell
o  hell
o
% test issue2039
[0;35;1;4m? foo[0m
[0;35;1;4mbar[0m
[0;35;1;4m? foo[0m
[0;35;1;4mbar.baz[0m
